### ISSUE TITLE
Non-error test output is sometimes not logged to the console unless -i is given

### ISSUE THREAD

**Issue by sjrd** [lihaoyi/mill#740]
I don't know what I did wrong in the support for Scala.js 1.0.0-RC1, but it is partially buggy. When running tests in a 1.0.0-RC1 project, the normal test log is not displayed on the console. For example, in https://github.com/lihaoyi/utest/pull/214, running `./mill utest.js[2.12.8,1.0.0-RC1].test` displays all the compiling log then ends with
```
Starting process: node
```
If a test is failing, the error summary is printed out:
```
Starting process: node
1 targets failed
utest.js[2.12.8,1.0.0-RC1].test.test test.utest.AssertsTests test.utest.AssertsTests.assert.success
```
and the exit code is correctly 1.

So at least it does not compromise the validity of sjs1.0-RC1 artifacts produced by mill 0.5.3. But it is very annoying for debugging.

When using the `-i` option to `mill`, the test output *is* correctly logged to the console. I have no idea what's different in that situation.

**Comment by lefou** [comment]
That's an issue that pops up in the chat regularly but until now, nobody opened an issue. It's related to the client server setup of mill, which is activated by default, and disabled by `-i`.

**Comment by lefou** [comment]
The output should pop up in `out/mill-worker-<...>/logs` though.

**Comment by lefou** [comment]
A minimal test case:

```scala
// build.sc
import mill._

def hello() = T.command {
  os.proc("echo", "hello").call(stdout = os.Inherit, stderr = os.Inherit)
  ()
}
```

```
$ mill -i hello
[1/1] hello
hello
$ mill hello
[1/1] hello
$ 
```

**Comment by lihaoyi** [comment]
Yeah if it's solved by `-i`, it's because subprocess output forwarding is an OS-level thing that isn't handled properly by the user-level stdout-redirection that Mill relies on.

Could probably be resolved by forwarding the Mill server's stdout to a file and having the client tailing that, rather than relyng on `System.setOut`, but would be a moderately pretty tricky refactor to make that happen

**Comment by lefou** [comment]
Strange thing is, that output goes either to STDOUT/STDERR or into `out/mill-worker-.../logs` but never to both. We looks like we handle them, but not properly.

**Comment by lefou** [comment]
_Just ignore my last comment._

**Comment by lihaoyi** [comment]
Yeah it's intentional-ish; we do a best effort to try and forward things on the server to the client's stdout/stderr, but for any stuff that we don't catch it's better it goes *somewhere* rather than just disappearing, so I chucked it into the `mill-worker-*/` folder.

I think the best thing to do is just make *everything* go to `mill-worker-*/logs`, and just have the mill client `tail` that in order to show the relevant logs to the user. There's some subtlety to it, e.g. we probably need separate `mill-worker-*/stdout`  and `mill-worker-*/stderr` files so the client can preserve the distinction between stderr/stdout, but I'm pretty sure we could make it work and I don't see any way to resolve this.

The fact that subprocess stdout/stderr inheritance doesn't follow user-land redirection isn't even a Scala/JVM thing; the same issue has bitten my in Python programs as well

**Comment by lefou** [comment]
I'm not sure I completely understand the issue. The out/../logs file is written by `MillClientMain`, which is on the client side. So the sub-process redirection (which happens on the server side) isn't the issue here. I just changed `MillClientMain` to directly output the out/err from the socket communication instead of writing it to a file, and it looks like it's working as expected on my machine with the simple repro I posted some comments before. I'd like to create a PR for your review and for further tests on different OSes.
